core/src/main/java/discord4j/core/object/entity/channel/ForumChannel.java

```diff
@@ -0,0 +1,312 @@
+package discord4j.core.object.entity.channel;
+
+import discord4j.common.annotations.Experimental;
+import discord4j.core.GatewayDiscordClient;
+import discord4j.core.object.entity.ForumTag;
+import discord4j.core.object.reaction.DefaultReaction;
+import discord4j.core.spec.ForumChannelEditSpec;
+import discord4j.core.spec.StartThreadInForumChannelSpec;
+import discord4j.core.util.EntityUtil;
+import discord4j.discordjson.json.ChannelData;
+import reactor.core.publisher.Mono;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * A Discord guild channel that contains organized threads with labels.
+ */
+@Experimental
+public final class ForumChannel extends BaseTopLevelGuildChannel implements CategorizableChannel {
+
+    public ForumChannel(GatewayDiscordClient gateway, ChannelData data) {
+        super(gateway, data);
+    }
+
+    /**
+     * Gets the topic for this guild forum.
+     *
+     * @return the user defined topic for this guild forum channel (0-4096 characters).
+     */
+    public Optional<String> getTopic() {
+        return getData().topic().toOptional()
+            .flatMap(opt -> opt); // Value is present on guild forum type but nullable
+    }
+
+    /**
+     * Gets whether this channel is considered NSFW (Not Safe For Work).
+     *
+     * @return {@code true} if this channel is considered NSFW (Not Safe For Work), {@code false} otherwise.
+     */
+    public boolean isNsfw() {
+        return getData().nsfw().toOptional().orElse(false);
+    }
+
+    /* Represents guild forum channels flags */
+    @Experimental
+    public enum ForumChannelFlag {
```

Comment:
im a little confuse here.. mostly because the Flags are for channels in general BUT this only show in forums... then the flags need to be here or can be moved to Channel...... if we think about the flags can grow for another things

Reply:
I think I am going to move the Enum into the Channel class because flags are defined as "global" channel flags as you said and not specifically ForumChannel flags.

However, as today the ̀flags` only shows in Forum channels, I'm not going to create a member in any parent interface and let it in ForumChannel.

Reply:
yeah in the future if discord add flags for more channels types then can consider make a method in Channel and if necesary use Overrides